clients/src/main/java/org/apache/kafka/clients/consumer/internals/Fetcher.java

```diff
@@ -649,10 +722,12 @@ private void handleListOffsetResponse(Map<TopicPartition, Long> timestampsToSear
         }
 
         // create the fetches
-        Map<Node, FetchRequest> requests = new HashMap<>();
+        Map<Node, FetchRequest.Builder> requests = new HashMap<>();
         for (Map.Entry<Node, LinkedHashMap<TopicPartition, FetchRequest.PartitionData>> entry : fetchable.entrySet()) {
             Node node = entry.getKey();
-            FetchRequest fetch = new FetchRequest(this.maxWaitMs, this.minBytes, this.maxBytes, entry.getValue());
+            FetchRequest.Builder fetch = new FetchRequest.Builder(
+                    this.maxWaitMs, this.minBytes, entry.getValue()).
```

Comment:
I'm a little unclear on the pattern for which fields are included in the builder constructor and which are passed as arguments. I thought perhaps it would be the required arguments included in the constructor, but we didn't pass the timestamps to query in the `ListOffsetRequest` above, which seems required.